components/supervisor/pkg/ports/exposed-ports.go

```diff
@@ -153,20 +155,24 @@ func (g *GitpodExposedPorts) doExpose(req *exposePortRequest) {
 		}
 		close(req.done)
 	}()
-	delay := g.minExposeDelay
 	attempt := 0
+	base := float64(g.minExposeDelay)
 	for {
 		_, err = g.C.OpenPort(req.ctx, g.WorkspaceID, req.port)
-		if err == nil || req.ctx.Err() != nil || attempt == 5 {
+		if err == nil || req.ctx.Err() != nil || attempt == int(g.maxExposeAttempts) {
 			return
 		}
-		log.WithError(err).WithField("port", req.port).Warnf("cannot expose port, trying again in %d seconds...", uint32(delay.Seconds()))
+		delay := time.Duration(base + rand.Float64()*(base*math.Pow(g.exposeDelayGrowFactor, float64(attempt))))
```

Comment:
I search code, it looks like we don't init rand seed in supervisor, so maybe it a pseudo random

Reply:
I will switch tomorrow to the library as suggested.